FGEGraphics/UISystem/UIBox.cs

```diff
@@ -0,0 +1,80 @@
+//
+// This file is part of the Frenetic Game Engine, created by Frenetic LLC.
+// This code is Copyright (C) Frenetic LLC under the terms of a strict license.
+// See README.md or LICENSE.txt in the FreneticGameEngine source root for the contents of the license.
+// If neither of these are available, assume that neither you nor anyone other than the copyright holder
+// hold any right or permission to use this software until such time as the official license is identified.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FGECore.MathHelpers;
+using FGEGraphics.ClientSystem;
+using FGEGraphics.GraphicsHelpers;
+using FGEGraphics.GraphicsHelpers.Textures;
+using OpenTK;
+using OpenTK.Graphics.OpenGL4;
+using OpenTK.Mathematics;
+
+namespace FGEGraphics.UISystem;
+
+/// <summary>Represents a simple box on the screen.</summary>
+/// <remarks>Constructs a simple colored box.</remarks>
+/// <param name="style">The style of the element.</param>
+/// <param name="pos">The position of the element.</param>
+/// <param name="shouldRender">Whether the element should render automatically.</param>
+public class UIBox(UIElementStyle style, UIPositionHelper pos, bool shouldRender = true) : UIElement(pos, shouldRender)
+{
+    /// <summary>The style of the box.</summary>
+    public UIElementStyle Style = style;
+    
+    // TODO: put this on UIPositionHelper or something
+    /// <summary>Whether this box is vertically flipped.</summary>
+    public bool Flip = false;
+
+    /// <summary>Returns the <see cref="Style"/>.</summary>
+    public override UIElementStyle GetStyle() => Style;
+
+    /// <summary>Renders this box on the screen.</summary>
+    /// <param name="view">The UI view.</param>
+    /// <param name="delta">The time since the last render.</param>
+    /// <param name="style">The current element style.</param>
+    public override void Render(ViewUI2D view, double delta, UIElementStyle style)
+    {
+        Vector3 rotation = new(-0.5f, -0.5f, LastAbsoluteRotation);
+        bool any = style.DropShadowLength > 0 || style.BorderColor.A > 0 || style.BaseColor.A > 0;
+        if (any)
+        {
+            Engine.Textures.White.Bind();
+        }
+        if (style.DropShadowLength > 0)
```

Comment:
All of these can/should be inside of the `any` check

Reply:
Sorry, can you elaborate?

Reply:
there's an
```
if (any) { .. }
if (thingThatOnlyHappensif any) { ... }
if (thingThatOnlyHappensif any) { ... }
if (thingThatOnlyHappensif any) { ... }
if (any) { ... }
```
it could be
```
if (any)
{
  ..
  if (thingThatOnlyHappensif any) { ... }
  if (thingThatOnlyHappensif any) { ... }
  if (thingThatOnlyHappensif any) { ... }
  ..
}
```